### PR TITLE
Update Package.md and Module.md 

### PR DESCRIPTION
**Pull request by kaneeldias**
## Purpose
$subject

Fixes https://github.com/ballerina-platform/ballerina-standard-library/issues/1627

## Checklist
- [x] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

**Review by niveathika — APPROVED** [review]

**Review by daneshk — APPROVED** [review]

File: mssql-ballerina/Module.md [hunk 1]
```diff
@@ -0,0 +1,391 @@
...
+This module provides the functionality required to access and manipulate data stored in a MSSQL database.
+
+### Prerequisite
+Add the MSSQL driver JAR as a native library dependency in your Ballerina project's `Ballerina.toml` file.
+It is recommended to use a MSSQL driver version greater than 9.2.0 as this module uses the database properties
+from the MSSQL driver version 9.2.0 onwards.
+
+Follow one of the following ways to add the JAR in the file:
+
+* Download the JAR and update the path
```
Comment (praneesha) [inline mssql-ballerina/Module.md:12]:
```suggestion
* Download the JAR and update the path.
```

File: mssql-ballerina/Module.md [hunk 2]
```diff
@@ -0,0 +1,391 @@
...
+    ```
+    [[platform.java11.dependency]]
+    path = "PATH"
+    ```
+
+* Add JAR with a maven dependency params
```
Comment (praneesha) [inline mssql-ballerina/Module.md:18]:
```suggestion
* Add JAR with the Maven dependency params.
```

File: mssql-ballerina/Module.md [hunk 3]
```diff
@@ -0,0 +1,391 @@
...
+    ```
+    [[platform.java11.dependency]]
+    groupId = "com.microsoft.sqlserver"
+    artifactId = "mssql-jdbc"
+    version = "9.2.0.jre11"
+    ```
+
+### Client
+To access a database, you must first create a
+[mssql:Client](https://docs.central.ballerina.io/ballerinax/mssql/latest/clients/Client) object.
```
Comment (praneesha) [inline mssql-ballerina/Module.md:28]:
```suggestion
[`mssql:Client`](https://docs.central.ballerina.io/ballerinax/mssql/latest/clients/Client) object.
```

File: mssql-ballerina/Module.md [hunk 4]
```diff
@@ -0,0 +1,391 @@
...
+    }
+  },
+  loginTimeout: 10
+};
+
+mssql:Client|sql:Error dbClient = new (user = "rootUser", password = "rootPass", options = mssqlOptions);
+```
+
+Similarly, the `dbClient` uses the named params and it provides an unshared connection pool of the type of
+[sql:ConnectionPool](https://docs.central.ballerina.io/ballerina/sql/latest/records/ConnectionPool)
```
Comment (praneesha) [inline mssql-ballerina/Module.md:67]:
```suggestion
[`sql:ConnectionPool`](https://docs.central.ballerina.io/ballerina/sql/latest/records/ConnectionPool)
```

File: mssql-ballerina/Module.md [hunk 5]
```diff
@@ -0,0 +1,391 @@
...
+       password: "password"
+    }
+  },
+  loginTimeout: 10
+};
+
+mssql:Client|sql:Error dbClient = new (user = "rootUser", password = "rootPass", options = mssqlOptions);
+```
+
+Similarly, the `dbClient` uses the named params and it provides an unshared connection pool of the type of
```
Comment (praneesha) [inline mssql-ballerina/Module.md:66]:
```suggestion
Similarly, the `dbClient` uses the named params and it provides an unshared connection pool of the 
```

File: mssql-ballerina/Module.md [hunk 6]
```diff
@@ -0,0 +1,391 @@
...
+[sql:ConnectionPool](https://docs.central.ballerina.io/ballerina/sql/latest/records/ConnectionPool)
+to be used within the client.
```
Comment (praneesha) [inline mssql-ballerina/Module.md:68]:
```suggestion
type to be used within the client.
```

File: mssql-ballerina/Module.md [hunk 7]
```diff
@@ -0,0 +1,391 @@
...
+For more details about connection pooling, see the [`sql` Module](https://docs.central.ballerina.io/ballerina/sql/latest).
+
+```ballerina
+mssql:Client|sql:Error dbClient = new (user = "rootUser", password = "rootPass", 
+                                       connectionPool = {maxOpenConnections: 5});
+```
+
+#### Using SSL
+To connect the MSSQL database using an SSL connection, you must add the SSL configurations to the `mssql:Options` when creating the `dbClient`.
```
Comment (praneesha) [inline mssql-ballerina/Module.md:77]:
```suggestion
To connect to the MSSQL database using an SSL connection, you must add the SSL configurations to the `mssql:Options` when creating the `dbClient`.
```

File: mssql-ballerina/Module.md [hunk 8]
```diff
@@ -0,0 +1,391 @@
...
+    cert: {
+        path: turstStorePath,
+        password: "password"
+    }
+  }
+};
+```
+#### Connection Pool Handling
+
+All ballerina database modules share the same connection pooling concept and there are three possible scenarios for
```
Comment (praneesha) [inline mssql-ballerina/Module.md:103]:
```suggestion
All Ballerina database modules share the same connection pooling concept and there are three possible scenarios for
```

File: mssql-ballerina/Module.md [hunk 9]
```diff
@@ -0,0 +1,391 @@
...
+connection pool handling.  For its properties and possible values, see the [`sql:ConnectionPool`](https://docs.central.ballerina.io/ballerina/sql/latest/records/ConnectionPool).
+
+1. Global shareable default connection pool
```
Comment (praneesha) [inline mssql-ballerina/Module.md:106]:
```suggestion
1. Global, shareable default connection pool
```

File: mssql-ballerina/Module.md [hunk 10]
```diff
@@ -0,0 +1,391 @@
...
+   If you define the `connectionPool` field inline when creating the database client with the `sql:ConnectionPool` type,
+   an unsharable connection pool will be created.
+
+    ```ballerina
+    mssql:Client|sql:Error dbClient = new ("localhost", "rootUser", "rootPass", connectionPool = { maxOpenConnections: 5 });
+    ```
+
+3. Local, shareable connection pool
+
+   If you create a record of type `sql:ConnectionPool` and reuse that in the configuration of multiple clients,
```
Comment (praneesha) [inline mssql-ballerina/Module.md:126]:
```suggestion
   If you create a record of `sql:ConnectionPool` type and reuse that in the configuration of multiple clients,
```

File: mssql-ballerina/Module.md [hunk 11]
```diff
@@ -0,0 +1,391 @@
...
+                               new (url = "localhost", "rootUser", "rootPass",
+                               connectionPool = connPool);
+    ```
+
+For more details about each property, see the [`mssql:Client`](https://docs.central.ballerina.io/ballerinax/mssql/latest/clients/Client) constructor.
+
+The [mssql:Client](https://docs.central.ballerina.io/ballerinax/mssql/latest/clients/Client) references
+[sql:Client](https://docs.central.ballerina.io/ballerina/sql/latest/clients/Client) and all the operations
+defined by the `sql:Client` will be supported by the `mdsql:Client` as well.
+defined by the `sql:Client` will be supported by the `mdsql:Client` as well.
```
Comment (praneesha) [inline mssql-ballerina/Module.md:149]:
```suggestion
defined by the `sql:Client` will be supported by the `mdsql:Client` as well.
```
Why is this line repeated?

File: mssql-ballerina/Module.md [hunk 12]
```diff
@@ -0,0 +1,391 @@
...
+#### Creating Tables
+
+This sample creates a table with two columns. One column is of type `int` and the other is of type `varchar`.
+The `CREATE` statement is executed via the `execute` remote function of the client.
+
+```ballerina
+// Create the ‘Students’ table with the  ‘id’, 'name', and ‘age’ fields.
+sql:ExecutionResult result = check dbClient->execute("CREATE TABLE student(id INT AUTO_INCREMENT, " +
+                         "age INT, name VARCHAR(255), PRIMARY KEY (id))");
+//A value of the`sql:ExecutionResult` type is returned for 'result'.
```
Comment (praneesha) [inline mssql-ballerina/Module.md:179]:
```suggestion
//A value of the`sql:ExecutionResult` type is returned for the 'result'. 
```

File: mssql-ballerina/Module.md [hunk 13]
```diff
@@ -0,0 +1,391 @@
...
+remote function.
+
+```ballerina
+sql:ExecutionResult result = check dbClient->execute("INSERT INTO student(age, name) " +
+                         "values (23, 'john')");
+```
+
+In this sample, the parameter values, which are in local variables are used to parameterize the SQL query in
+the `execute` remote function. This type of parameterized SQL query can be used with any primitive Ballerina type
+like `string`, `int`, `float`, or `boolean` and in that case, the corresponding SQL type of the parameter is derived
```
Comment (praneesha) [inline mssql-ballerina/Module.md:197]:
```suggestion
like `string`, `int`, `float`, or `boolean`. In that case, the corresponding SQL type of the parameter is derived
```

File: mssql-ballerina/Module.md [hunk 14]
```diff
@@ -0,0 +1,391 @@
...
+string name = "Anne";
+int age = 8;
+
+sql:ParameterizedQuery query = `INSERT INTO student(age, name)
+                                values (${age}, ${name})`;
+sql:ExecutionResult result = check dbClient->execute(query);
+```
+
+In this sample, the parameter values are passed as a `sql:TypedValue` to the `execute` remote function. Use the
+corresponding subtype of the `sql:TypedValue` such as `sql:Varchar`, `sql:Char`, `sql:Integer`, etc., when you need to
```
Comment (praneesha) [inline mssql-ballerina/Module.md:210]:
```suggestion
corresponding subtype of the `sql:TypedValue` such as `sql:Varchar`, `sql:Char`, `sql:Integer`, etc. when you need to
```

File: mssql-ballerina/Module.md [hunk 15]
```diff
@@ -0,0 +1,391 @@
...
+
+#### Querying Data
+
+These samples show how to demonstrate the different usages of the `query` operation and query the
+database table and obtain the results.
+
+This sample demonstrates querying data from a table in a database.
+First, a type is created to represent the returned result set. This record can be defined as an open or a closed record
+according to the requirement. If an open record is defined, the returned stream type will include both defined fields
+in the record and additional database columns fetched by the SQL query which are not defined in the record.
```
Comment (praneesha) [inline mssql-ballerina/Module.md:249]:
```suggestion
in the record and additional database columns fetched by the SQL query, which are not defined in the record.
```

File: mssql-ballerina/Module.md [hunk 16]
```diff
@@ -0,0 +1,391 @@
...
+Note the mapping of the database column to the returned record's property is case-insensitive if it is defined in the
+record(i.e., the `ID` column in the result can be mapped to the `id` property in the record). Additional Column names
```
Comment (praneesha) [inline mssql-ballerina/Module.md:251]:
```suggestion
record(i.e., the `ID` column in the result can be mapped to the `id` property in the record). Additional column names are
```

File: mssql-ballerina/Module.md [hunk 17]
```diff
@@ -0,0 +1,391 @@
...
+added to the returned record as in the SQL query. If the record is defined as a close record, only defined fields in the
```
Comment (praneesha) [inline mssql-ballerina/Module.md:252]:
```suggestion
added to the returned record as in the SQL query. If the record is defined as a close record, only the defined fields in the
```

File: mssql-ballerina/Module.md [hunk 18]
```diff
@@ -0,0 +1,391 @@
...
+record are returned or gives an error when additional columns present in the SQL query. Next, the `SELECT` query is executed
```
Comment (praneesha) [inline mssql-ballerina/Module.md:253]:
```suggestion
record are returned or gives an error when additional columns are present in the SQL query. Next, the `SELECT` query is executed
```

File: mssql-ballerina/Module.md [hunk 19]
```diff
@@ -0,0 +1,391 @@
...
+via the `query` remote function of the client. Once the query is executed, each data record can be retrieved by looping
+the result set. The `stream` returned by the select operation holds a pointer to the actual data in the database and it
```
Comment (praneesha) [inline mssql-ballerina/Module.md:255]:
```suggestion
the result set. The `stream` returned by the `select` operation holds a pointer to the actual data in the database and it
```

File: mssql-ballerina/Module.md [hunk 20]
```diff
@@ -0,0 +1,391 @@
...
+// Define an open record type to represent the results.
+type Student record {
+    int id;
+    int age;
+    string name;
+};
+
+// Select the data from the database table. The query parameters are passed 
+// directly. Similar to the `execute` samples, parameters can be passed as
+// sub types of `sql:TypedValue` as well.
```
Comment (praneesha) [inline mssql-ballerina/Module.md:268]:
```suggestion
// sub types of the `sql:TypedValue` as well.
```

File: mssql-ballerina/Module.md [hunk 21]
```diff
@@ -0,0 +1,391 @@
...
+        sql:Error? e = resultStr.forEach(function(record{} result) {
+        //can perform operations using 'result'.
+      });
+    }
+    check result.close();
+}
+```
+Note that you have to explicitly invoke the close operation on the `sql:ProcedureCallResult` to release the connection resources and avoid a connection leak as shown above.
+
+>**Note:** The default thread pool size used in Ballerina is: `the number of processors available * 2`. You can configure the thread pool size by using the `BALLERINA_MAX_POOL_SIZE` environment variable.
```
Comment (praneesha) [inline mssql-ballerina/Module.md:391]:
```suggestion
>**Note:** The default thread pool size used in Ballerina is: `the number of processors available * 2`. You can configure the thread pool size by using the `BALLERINA_MAX_POOL_SIZE` environment variable.
```
If we are adding this format for the Note here, we need to be consistent and add the same to all the other occurrences as well.

**Comment by niveathika** [comment]
@kaneeldias Let's address the comments and update the PR

**Pull request merged**